src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -934,48 +941,45 @@ public async Task<string> PayInvoiceAfterSubscriptionChangeAsync(ISubscriber sub
                 {
                     PaymentMethod = cardPaymentMethodId,
                 };
-                if (invoice.AmountDue > 0)
+                if (customer?.Metadata?.ContainsKey("btCustomerId") ?? false)
```

Comment:
This is really annoying formatting. All this is is removing an `if` and shifting over by 1 tab.

Reply:
Didn't we check that the key exists on line 903?

Reply:
Yes, the logic for a braintree charge is split into two locations (this and 903). I don't want to refactor the way that this works right now. I'd rather approach all these spaghetti issues more methodically once we can test well.